### PR TITLE
added log record with investment address to prevent duplicate mint

### PR DESCRIPTION
**Pull request by niponchi**

### CODE REVIEW

File: src/mintnft/mint.ts [hunk 1]
```diff
@@ -57,10 +66,19 @@ export namespace MintNFT {
...
     to: PublicKey
   ): Promise<TransactionInstruction[]> {
     const mintRent = await connection.getMinimumBalanceForRentExemption(MintLayout.span)
+    const loggerProgram = newLoggerProgram(PDA_LOGGER_PROGRAM_ID, connection)
+
+    const mintLogAddress = await findMintLogProgramAddress(uniqueKey, walletPub)
+    const mintLogData = await loggerProgram.account.logEntry.fetchNullable(mintLogAddress)
+
+    if (mintLogData) {
+      throw new Error('token_already_minted')
```
Comment (zapkub) [inline src/mintnft/mint.ts:80]: should be `ForbiddenError` see
https://github.com/jshttp/http-errors#list-of-all-constructors

File: src/pda-logger/index.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Program, Provider, Wallet, web3 } from '@project-serum/anchor'
+import { Logger, IDL as LoggerIdl } from './logger'
+import { Connection, PublicKey } from '@solana/web3.js'
+
+export const newLoggerProgram = (programId: PublicKey, connection: Connection): Program<Logger> => {
```
Comment (zapkub) [inline src/pda-logger/index.ts:5]: This project based on `namespace` pattern. This module should be also implemented under some namespace

eg.

```typescript
export namespace PDALogger {
     export const newProgram() => {}
}
```

**Review by zapkub — CHANGES_REQUESTED** [review]

File: src/mintnft/mint.ts (hunk 1 above)
Comment (niponchi) [reply src/mintnft/mint.ts]: Updated

File: src/pda-logger/index.ts (hunk 2 above)
Comment (niponchi) [reply src/pda-logger/index.ts]: updated

**Comment by zapkub** [comment]
Cool, didnt test this yet but LGTM

**Comment by zapkub** [comment]
close https://github.com/avareum/avareum-nft/issues/4

**Pull request merged**